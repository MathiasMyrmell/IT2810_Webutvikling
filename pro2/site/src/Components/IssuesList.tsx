import { IssuesContext } from "./Issues";
import './IssuesList.css';

function valueToTable(values: any) { // Converts object to two tables
    var openList;
    if (values.open.length == 0) {
        openList = (
            <tr key="1">
                <td>No open issues.</td>
            </tr>
        )
    } else {
        openList = values.open.map((item: any) => (
            <tr key={item.id}>
                <td>#{item.id}: {item.title}</td>
            </tr>
        ));
    }

    var closedList;
    if (values.closed.length == 0) {
        closedList = (
            <tr key="1">
                <td>No closed issues.</td>
            </tr>
        )
    } else {
        closedList = values.closed.map((item: any) => (
            <tr key={item.id}>
                <td>#{item.id}: {item.title}</td>
            </tr>
        ));
    }
    return (
        <div className="IssuesList">
            <table className="openIssues tableIssues">
                <tbody>
                    <tr key="header">
                        <th>Open</th>
                    </tr>
                    {openList}
                </tbody>
            </table>
            <table className="closedIssues tableIssues">
                <tbody>
                    <tr key="header">
                        <th>Closed</th>
                    </tr>
                    {closedList}
                </tbody>
            </table>
        </div>
    );
}

export default function IssuesList() {
    return (
        // Here we get the data
        <IssuesContext.Consumer>
            {value => valueToTable(value)}
        </IssuesContext.Consumer>

    )
}