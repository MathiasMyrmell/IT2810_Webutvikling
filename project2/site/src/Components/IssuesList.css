.IssuesList {
    display: flex;
    flex-wrap: wrap;
}

.tableIssues {
    flex-grow: 1;
    height: 100%;
    width: 512px;
    margin: 31px;
}

.tableIssues td,
.tableIssues th {
    border: 1px solid #ddd;
    padding: 10px;
    font-size: clamp(6px, 2vw, 20px);
}

.tableIssues tr:nth-child(even) {
    background-color: #f2f2f2;
}

.tableIssues tr:hover {
    background-color: #ddd;
}

.tableIssues th:hover {
    background-color: rgb(66, 70, 78);
}

.tableIssues th {
    padding-top: 12px;
    padding-bottom: 12px;
    text-align: left;
    background-color: rgb(50, 54, 62);
    color: rgb(137, 227, 252);
    font-size: clamp(14px, 1.5vw, 20px);
}

.tableIssues td {
    font-size: clamp(14px, 1.5vw, 20px);
}

/* For small screens */
@media (max-width: 1200px) {

    .tableIssues {
        width: 100%;
        margin: 0px;
    }

}