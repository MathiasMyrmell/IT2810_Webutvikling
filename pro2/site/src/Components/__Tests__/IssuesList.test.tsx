import renderer from 'react-test-renderer';
import IssuesList from '../IssuesList';

test('Snapshot test IssuesList', () => {
    const tree = renderer.create(<IssuesList />).toJSON();
    expect(tree).toMatchSnapshot();
});
