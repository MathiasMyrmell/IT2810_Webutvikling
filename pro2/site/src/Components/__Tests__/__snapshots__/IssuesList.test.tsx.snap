// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test IssuesList 1`] = `
<div
  className="IssuesList"
>
  <table
    className="openIssues tableIssues"
  >
    <tbody>
      <tr>
        <th>
          Open
        </th>
      </tr>
      <tr>
        <td>
          No open issues.
        </td>
      </tr>
    </tbody>
  </table>
  <table
    className="closedIssues tableIssues"
  >
    <tbody>
      <tr>
        <th>
          Closed
        </th>
      </tr>
      <tr>
        <td>
          No closed issues.
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
